### PR TITLE
Add endpoint to add new test #2

### PR DESCRIPTION
**Pull request by didasmbalanya**
# PR Details
 - add migration to create answers table
 - add jest for tests

## Description
A user can post a new Test with the answers and their interpretation

## How Has This Been Tested

checkout to `ft-add-a-Test-endpoint-#2`
set up a database for the project
set a `.env` file based on the variables on `.env.sample`
run `npm install`
run `npm run db:migrate`
run `npm run dev` to start app
send a POST request to `http://localhost:8000/api/v1/tests', a sample of body attached bellow
 - 
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Screenshots
Screen Shot 2020-02-19 at 12 33 51

### CODE REVIEW

File: src/controllers/TestController.js [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
-class TestController { }
+class TestController { 
+  static async addNewTest(req, res){
+    const { category, questions, title, answers } = req.body;
+
+    const result = await addnewTestToDb(category, questions, title, answers);
+    if (result.errors) return res.status(422).send({ status: 422, error: result.errors[0].message})
+    
+    result.status = 200;
+    res.status(201).send(result);
```
Comment (cop1fab) [inline src/controllers/TestController.js]: Let's return messages.

File: src/database/models/test.js [hunk 2]
```diff
@@ -13,6 +14,9 @@ export default (sequelize, DataTypes) => {
     Test.hasMany(models.Question, {
       foreignKey: 'testId'
     })
+    Test.hasMany(models.Answer, {
+      foreignKey: 'testId'
+    })
```
Comment (cop1fab) [inline src/database/models/test.js]: clean code convention ......

File: src/services/testService.js [hunk 3]
```diff
@@ -1,10 +1,45 @@
-import database from '../database/models';
+import database from "../database/models";
 /**
  * @class TestServices
  */
 
- class TestServices{
+class TestServices {
+  static async addnewTestToDb(category, questions, title, answers) {
```
Comment (cop1fab) [inline src/services/testService.js]: Please pay attention on the camel-case naming convention.

File: src/database/models/test.js (hunk 2 above)
Comment (didasmbalanya) [reply src/database/models/test.js]: I am not sure what exactly you mean here...

File: src/services/testService.js (hunk 3 above)
Comment (didasmbalanya) [reply src/services/testService.js]: That's a class. Classes use Pascal naming for JS

Comment (didasmbalanya) [reply src/services/testService.js]: I have changed the service name to addNewTestService

**Pull request merged**